src/components/EditorNav.tsx

```diff
@@ -51,6 +52,7 @@ export default function EditorNav({
   onSave: () => void
 }) {
   const [, navigate] = useLocation()
+  const { circuitJson } = useRunTsx(code, snippet?.snippet_type)
```

Comment:
can you avoid using `useRunTsx`? It's a very expensive hook and the circuitJson should be computed elsewhere